ddtrace/internal/debug.py

```diff
@@ -123,12 +123,14 @@ def collect(tracer):
         tracer_enabled=tracer.enabled,
         sampler_type=type(tracer.sampler).__name__ if tracer.sampler else "N/A",
         priority_sampler_type=type(tracer.priority_sampler).__name__ if tracer.priority_sampler else "N/A",
+        sampler_rules=ddtrace.config.sampler_rules,
```

Comment:
The issue with using a config setting like this is that if they manually configure a sampler without the env variables then we will not get the proper rules.

Instead we might want to consider something like:

```python
sampler_rules = None

if isinstance(tracer.sampler, DatadogSampler):
    sampler_rules = [str(rule) for rule in tracer.sampler.rules]

return dict(
    ...
    sampler_rules=sampler_rules,
)
```


We could also consider doing the same with the `DatadogSampler.limiter`

```python
sampler_limiter = None
if isinstance(tracer.sampler, DatadogSampler):
    sampler_limiter = str(tracer.sampler.limiter)

return dict(
    ...
    sampler_limiter=sampler_limiter,
)
```


... or... we could consider add a `__str__`/`__repr__` onto each of our sampler classes so we can just do:

```python
return dict(
    ...
    sampler=str(tracer.sampler),
)
```

To get something like `sampler="DatadogSampler(limiter=RateLimiter(rate_limit=100, ...), rules=[SamplingRule(sample_rate=1.0, service="my-srvc", ...), ...])"`.

Going this route could make it easier than basically unpacking all the metadata from a given sampler.

What do you think?

Reply:
I went with you first suggestion but if you feel that the second route is cleaner I'm happy to implement it